src/Modules/OrchardCore.Commerce/Extensions/UserManagerExtensions.cs

```diff
@@ -12,8 +12,16 @@ namespace Microsoft.AspNetCore.Identity;
 public static class UserManagerExtensions
 {
     /// <summary>
-    /// Returns the custom user setting of <see cref="UserAddressesPart"/> for the given <paramref name="principal"/>.
+    /// Returns the custom user setting of <see cref="UserAddressesPart"/> for the given <paramref name="principal"/>
+    /// if the corresponding <see cref="User"/> is not <see langword="null"/>, returns <see langword="null"/> otherwise.
     /// </summary>
     public static async Task<UserAddressesPart> GetUserAddressAsync(this UserManager<IUser> userManager, ClaimsPrincipal principal) =>
         await userManager.GetUserAsync(principal) is User user ? user.As<ContentItem>(UserAddresses)?.As<UserAddressesPart>() : null;
+
+    /// <summary>
+    /// Returns the custom user setting of <see cref="UserDetailsPart"/> for the given <paramref name="principal"/>
+    /// if the corresponding <see cref="User"/> is not <see langword="null"/>, returns <see langword="null"/> otherwise.
+    /// </summary>
+    public static async Task<UserDetailsPart> GetUserDetailsAsync(this UserManager<IUser> userManager, ClaimsPrincipal principal) =>
+        await userManager.GetUserAsync(principal) is User user ? user.As<ContentItem>(UserDetails)?.As<UserDetailsPart>() : null;
```

Comment:
If you write `(await userManager.GetUserAsync(principal) as User)?` instead of `user` then the conditional expression can be avoided.

Reply:
`(await userManager.GetUserAsync(principal) as User).As<ContentItem>(UserDetails)?.As<UserDetailsPart>();` can here indeed be a shortcut as you don't need the instance of the `User` type in this class.

Reply:
Same applies for the method above this one.